File: Uler/Hospital.py
from Doctor import Doctor
from Patient import Patient

class Hospital:
    def __init__(self, name):
        self.hospital_name = name
        self.patients = []
        self.doctors = []

    # getter & setter
    def set_hospital_name(self, name):
        self.hospital_name = name

    def get_hospital_name(self):
        return self.hospital_name

    # fungsi nambahin pasien dan dokter
    def add_patient(self, patient: Patient):
        self.patients.append(patient)

    def add_doctor(self, doctor: Doctor):
        self.doctors.append(doctor)

    # fungsi print
    def print_info(self):
        print(f"=>> RSUD {self.hospital_name} <<=")

        print("\nList Doctors:\n")
        for d in self.doctors:
            d.print_doctor()

        print("\nList Patients:\n")
        for p in self.patients:
            p.print_patient()

File: CPP/Code/Hospital.cpp
//manggil kelas
#include "Person.cpp"
#include "Schedule.cpp"
#include "Doctor.cpp"
#include "Medicine.cpp"
#include "Patient.cpp"
#include <iostream>
#include <vector>
using namespace std;

class Hospital {
private:
    string hospitalName;
    vector<Patient> patients;
    vector<Doctor> doctors;

public:

//getter setter
    Hospital(string name) {
        hospitalName = name;
    }

    void setHospitalName(string name) { 
      hospitalName = name; 
    }

    string getHospitalName() const { 
      return hospitalName; 
    }

    //fungsi nambahin pasien dan dokter
    void addPatient(const Patient& p) {
       patients.push_back(p); 
    }
    void addDoctor(const Doctor& d) { 
      doctors.push_back(d); 
    }

    //fungsi print
    void printInfo() {
        cout << "=>> RSUD " << hospitalName << " <<=" << endl;

        cout << "\nList Doctors:\n" << endl;
        for (auto& d : doctors) {
            d.printDoctor();
        }

        cout << "\nList Patients:\n" << endl;
        for (auto& p : patients) {
            p.printPatient();
        }
    }
};